project-icp/bee-pollinator-app
# Integrate initial pollinator model

#### Overview
Adds an initial version of the pollinator model to the app and makes it callable by the front end.  It also removes a considerable amount of modelling/geoprocessing code that was needed for MMW but is not for this project.  I decided to create the model as a python module, but one that was embedded in this repo.  I'm hoping that this will reduce the feedback cycle during development as opposed to having it in its own repo with a deployment process.  The module could be plucked out in the future when we want a properly versioned and separate product.  

#### Notes
* This is a big PR, with a lot of files getting touched.  My recommendation for review is:
    * The 3 commits are pretty well isolated, I think it makes sense to evaluate them in sequence
    * The majority of the changes are wholesale deletions
    * There is a chunk of code for adding a python `pollinator` module.  This is fairly initial, and I wouldn't recommend too much time reviewing the _implementation_ of that code, but more perhaps the structure and how it is included in the overall project.  I anticipate near term work to copy over some tests add comments etc, which will probably be a better time to review and evaluate the implementation
* The model is presently returning a fairly meaningless number, but it's derived from an interim stage of the model (farm abundance).
* I tried to strike a balance between removing unneeded code and just letting the existing (and quicker to implement) code from MMW function.  For example, there is only one model "package" now, but I left a lot of code that assumed there may be multiple.

Connects #53 
Connects https://github.com/project-icp/bee-pollinator-model/issues/3

#### Testing
* Get the raster data from @mmcfarland
* Bring up your vagrant machines
* Ensure that you have created the initial model tables (`./scripts/manage.sh migrate`)
* `vagrant provision worker`
* Choose a small AoI (< 400 acres) and create a model project
* In DevTools verify that the initial `yield` endpoint is requested and that after polling, a numerical result is returned.  This applies to each scenario and after modifications are made
